assets/javascripts/discourse/controllers/group-reports-show.js

```diff
@@ -6,93 +6,111 @@ import Bookmark, {
+  @tracked results = this.model.results;
+  @tracked queryGroupBookmark = this.queryGroup?.bookmark;
 
-    toggleBookmark() {
-      return openBookmarkModal(
-        this.queryGroup.bookmark ||
-          Bookmark.create({
-            bookmarkable_type: "DataExplorer::QueryGroup",
-            bookmarkable_id: this.queryGroup.id,
-            user_id: this.currentUser.id,
-          }),
-        {
-          onAfterSave: (savedData) => {
-            const bookmark = Bookmark.create(savedData);
-            this.set("queryGroup.bookmark", bookmark);
-            this.appEvents.trigger(
-              "bookmarks:changed",
-              savedData,
-              bookmark.attachedTo()
-            );
-          },
-          onAfterDelete: () => {
-            this.set("queryGroup.bookmark", null);
-          },
-        }
-      );
-    },
+  explain = false;
 
-    // This is necessary with glimmer's one way data stream to get the child's
-    // changes of 'params' to bubble up.
-    updateParams(identifier, value) {
-      this.set(`model.params.${identifier}`, value);
-    },
-  }, // actions
+  get hasParams() {
+    return this.model.param_info.length > 0;
+  }
 
-  @discourseComputed("queryGroup.bookmark")
-  bookmarkLabel(bookmark) {
-    return bookmark ? "bookmarked.edit_bookmark" : "bookmarked.title";
-  },
+  get bookmarkLabel() {
+    return this.queryGroupBookmark
+      ? "bookmarked.edit_bookmark"
+      : "bookmarked.title";
+  }
 
-  @discourseComputed("queryGroup.bookmark")
-  bookmarkIcon(bookmark) {
-    if (bookmark && bookmark.reminder_at) {
+  get bookmarkIcon() {
+    if (this.queryGroupBookmark && this.queryGroupBookmark.reminder_at) {
       return WITH_REMINDER_ICON;
     }
     return NO_REMINDER_ICON;
-  },
+  }
 
-  @discourseComputed("queryGroup.bookmark")
-  bookmarkClassName(bookmark) {
-    return bookmark
+  get bookmarkClassName() {
+    return this.queryGroupBookmark
       ? ["query-group-bookmark", "bookmarked"].join(" ")
       : "query-group-bookmark";
-  },
-});
+  }
+
+  @bind
+  async run() {
+    this.loading = true;
+    this.showResults = false;
+
+    try {
+      const response = await ajax(
+        `/g/${this.get("group.name")}/reports/${this.model.id}/run`,
+        {
+          type: "POST",
+          data: {
+            params: JSON.stringify(this.model.params),
+            explain: this.explain,
+          },
+        }
+      );
+
+      this.results = response;
+      if (!response.success) {
+        return;
+      }
+      this.showResults = true;
+    } catch (error) {
+      if (
+        error.jqXHR &&
+        error.jqXHR.status === 422 &&
+        error.jqXHR.responseJSON
+      ) {
+        this.results = error.jqXHR.responseJSON;
+      } else {
+        popupAjaxError(error);
+      }
+    } finally {
+      this.loading = false;
+    }
+  }
+
+  @action
+  toggleBookmark() {
+    return openBookmarkModal(
+      this.queryGroupBookmark ||
+        Bookmark.create({
```

Comment:
the second target for `store.createRecord` 😉

Reply:
https://github.com/discourse/discourse-data-explorer/pull/218/commits/ea939f11627ceb33c221874a98140fde6d8f0e01 :)